tests/models/test_fracture_damage.py

```diff
@@ -0,0 +1,115 @@
+import copy
+from typing import Literal
+
+import numpy as np
+import pytest
+
+import porepy as pp
+from porepy.applications.md_grids.model_geometries import (
+    CubeDomainOrthogonalFractures,
+    SquareDomainOrthogonalFractures,
+)
+from porepy.applications.test_utils.models import _add_mixin
+from porepy.compositional.materials import FractureDamageSolidConstants
+from porepy.examples import fracture_damage as damage
+
+geometry_mixins = [
+    SquareDomainOrthogonalFractures,
+    CubeDomainOrthogonalFractures,
+]
+# Additional solid parameters used for parameterizing the tests for different regimes
+# of frictional and dilational damage.
+additional_solid_params = {
+    "dilation": {
+        "initial_friction_damage": 1.0,  # Initial value 1 implies no frictional damage
+        "initial_dilation_damage": 2.0,
+    },
+    "friction": {
+        "initial_friction_damage": 1.5,
+        "initial_dilation_damage": 1.0,  # Initial value 1 implies no dilational damage
+    },
+    "both": {
+        "initial_friction_damage": 1.1,
+        "initial_dilation_damage": 1.1,
+    },
+}
+
+
+@pytest.mark.parametrize("isotropic", [True, False])
+@pytest.mark.parametrize("dim", [2, 3])
+@pytest.mark.parametrize(
+    "regime",
+    ["dilation", "friction", "both"],
+)
+@pytest.mark.parametrize(
+    "time_steps", [2, pytest.param(7, marks=pytest.mark.skipped(reason="slow"))]
+)
+def test_damage(
+    isotropic: bool,
+    dim: int,
+    regime: Literal["dilation", "friction", "both"],
+    time_steps: int,
+):
+    # Copy the model parameters to avoid modifying the original dictionary.
+    # Deepcopy is needed to avoid modifying mutable objects in the dictionary,
+    # presumably the time manager.
+    params_local = copy.deepcopy(damage.model_params)
+    if isotropic:
+        # Use the isotropic damage model.
+        model_class = damage.IsotropicFractureDamage
+        # Change the exact solution to the isotropic version.
+        params_local["exact_solution"] = damage.ExactSolutionIsotropic
+    else:
+        # Use the anisotropic damage model.
+        model_class = damage.AnisotropicFractureDamage
+    # Add geometry mixin.
+    model_class = _add_mixin(geometry_mixins[dim - 2], model_class)
```

Comment:
I prefer a dictionary over a list of geometries here. This is too clever for me.